### ISSUE TITLE
What is the meaning of multiple annotations?

### ISSUE THREAD

**Issue by seanmcl** [typetools/checker-framework#3037]
What is the meaning of the following snippet, which typechecks:

```
    interface Consumer<@NonNull T> {
        default @Nullable T f() {
            return null;
        }
    }
```

I expected a type error here because multiple annotations were applied to the same type.  Here `NonNull` is a subtype of `Nullable`, so does CF infer something like the least upper bound of all the type annotations?

Thanks!

**Comment by wmdietl** [comment]
I don't see "multiple annotations" in the example.

In the first line `interface Consumer<@NonNull T>` we have the declaration of type parameter `T`.
An annotation on the type parameter declaration itself sets the lower bound of the type parameter. 
https://checkerframework.org/manual/#generics-bounds-syntax
The lower bound is already `@NonNull`, so this doesn't have any effect.
The upper bound of the type parameter is implicitly `@Nullable Object`.
A possible improvement here would be to warn on the redundant `@NonNull` annotation.

In the second line `@Nullable T` is an annotation on a type variable use.
https://checkerframework.org/manual/#type-variable-use
Such an annotation ignores the annotation from the corresponding type argument.
In this example, method `f` can always return `null`, even though `T` might be instantiated to be `@NonNull String`.

Does this make sense to you now? Is there anything we can improve in those manual sections?

**Comment by seanmcl** [comment]
Thank you for the fast response and links!

The manual makes the meaning clear

> The type annotation does not restrict how MyClass2 may be instantiated. In other words, both MyClass2<@NonNull String> and MyClass2<@Nullable String> are legal,

I just need to read the manual more and stop trying to guess the meaning of the annotations!  Like 
in #3038 I assumed the type variable annotation was exact, e.g. a lower and upper bound.

**Comment by vlsi** [comment]
@wmdietl , 

>wmdietl: An annotation on the type parameter declaration itself sets the lower bound of the type parameter.

This contradicts the documentation section https://checkerframework.org/releases/1.1.2/checkers-manual.html#generics which says:

```
For example, these two declarations have the same effect:

  class MyClassNN<@NonNull T> { ... }
  class MyClassNN<T extends @NonNull Object super @NonNull void> { ... }
```

Does that mean `checkers-manual.html#generics` is wrong and it needs to be updated?

**Comment by mernst** [comment]
@vlsi You have linked to the manual for Checker Framework version 1.1.2, which is 10 years old.
Could you please use the current version of the manual?  If it is confusing or incorrect, definitely let us know so that we can improve it.

I have made the URL redirect (this removes direct access to the ancient manual, but that is probably not a problem).

**Comment by vlsi** [comment]
Oh, I was trying to annotate a generic-heavy code, and I think I've bumped into 1.1.2 in the search results.
It would definitely help if you could set up a redirect or create an overlay like "you are viewing documentation for an outdated version".
